rfcs/0001-messages.md

```diff
@@ -0,0 +1,301 @@
+# FROST messages
+
+Proposes a message layout to exchange information between participants of a FROST setup using the [jubjub](https://github.com/zkcrypto/jubjub) curve.
+
+## Motivation
+
+Currently FROST library is complete for 2 round signatures with a dealer/aggregator setup.
+This proposal is only considering that specific features, additions and upgrades will need to be made when DKG is implemented.
+
+Assuming all participants have a FROST library available we need to define message structures in a way that data can be exchanged between participants. The proposal is a collection of data types so each side can do all the actions needed for a real life situation.
+
+## Definitions
+
+- `dealer`
+- `aggregator`
+- `signer`
+- `nonce`
+- `commitment`
+- 
+
+## Guide-level explanation
+
+We propose a message separated in 2 parts, a header and a payload:
+
+```rust
+struct Message {
+    header: Header,
+    payload: Payload,
+}
+```
+
+`Header` will look as follows:
+
+```rust
+struct Header {
+    msg_type: MsgType,
+    version: MsgVersion,
+    sender: ParticipantID,
+    receiver: ParticipantID,
+}
+```
+
+While `Payload` will be defined as:
+
+```rust
+enum Payload {
+    DealerBroadcast(MsgDealerBroadcast),
+    Commitments(MsgCommitments),
+    SigningPackage(MsgSigningPackage),
+    SignatureShare(MsgSignatureShare),
+    FinalSignature(MsgFinalSignature),
+}
+```
+
+All the messages and new types will be defined in a new file `src/frost/messages.rs`
+
+## Reference-level explanation
+
+Here we explore in detail the header types and all the message payloads. 
+
+### Header
+
+Fields of the header define new types. Proposed implementation for them is as follows:
+
+```rust
+#[repr(u8)]
+#[non_exhaustive]
+enum MsgType {
+    DealerBroadcast,
+    Commitments,
+    SigningPackage,
+    SignatureShare,
+    FinalSignature,
+}
+
+struct MsgVersion(u8);
+
+struct ParticipantID(u8);
+```
+
+### Payloads
+
+Each payload defines a new message:
+
+```rust
+/// Dealer must send this message with initial data to each participant involved.
+/// With this, the participant should be able to build a `SharePackage` and use
+///  the `sign()` function.
+/// `public_key` can be calculated from the `secret_key`.
+struct MsgDealerBroadcast {
+    /// The secret key as a frost::Scalar.
+    secret_key: frost::Scalar,
+    /// Commitment for the signer as a single jubjub::AffinePoint.
+    commitment: jubjub::AffinePoint,
+    /// The public signing key that represents the entire group.
+    group_public: GroupPublic,
+}
+
+/// The point and verification bytes needed to generate the group public key
+struct GroupPublic {
+    /// The point
+    point: jubjub::AffinePoint,
+    /// The verification bytes
+    bytes: [u8; 32],
```

Comment:
Let's give this field a more specific name, once we know which fields we need in this struct.